### PR TITLE
Support for MySQL and PostgreSQL - Production grade DB

### PR DESCRIPTION
**Pull request by gokulprathin8**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -79,6 +79,19 @@
 		</dependency>
 
 
+		<dependency>
+			<groupId>org.postgresql</groupId>
+			<artifactId>postgresql</artifactId>
+			<version>42.3.8</version>
```
Comment (ZihengSun) [inline pom.xml:85]: Are these versions the latest? It is ok if not, but need to watch if these versions are out of maintenance or don't work with the latest postgresql/my db.

File: src/main/java/com/gw/database/EnvironmentRepository.java [hunk 2]
```diff
@@ -37,7 +35,7 @@ Collection<Environment> findEnvByID_BIN_ENV_BaseDir(
    * @param bin The binary path.
    * @return A collection of environment configurations matching the specified host and binary path.
    */
-  @Query(value = "select * from environment where hostid = ?1 and bin = ?2", nativeQuery = true)
+  @Query("SELECT e FROM Environment e WHERE e.hostobj.id = :hostid AND e.bin = :bin")
```
Comment (ZihengSun) [inline src/main/java/com/gw/database/EnvironmentRepository.java:38]: Does this work with existing database records? I mean the records created before this PR>

File: src/main/java/com/gw/jpa/Environment.java [hunk 3]
```diff
@@ -22,11 +17,11 @@ public class Environment {
...
 	private String name, type, bin, pyenv, basedir;
 
-	@ManyToOne(fetch = FetchType.LAZY)
+	@ManyToOne(fetch = FetchType.EAGER)
 	@JoinColumn(name = "hostid")
 	@JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
 	private Host hostobj;
 
-	@Column(columnDefinition = "LONGTEXT")
+	@Lob
```
Comment (ZihengSun) [inline src/main/java/com/gw/jpa/Environment.java:25]: Will this work with existing records as I mentioned? Or do people need to do db migration first?

File: src/main/java/com/gw/jpa/GWUser.java [hunk 4]
```diff
@@ -1,55 +1,40 @@
...
 @Getter
 @Setter
 @NoArgsConstructor
-@RequiredArgsConstructor
 public class GWUser {
 
     @Id
     private String id;
 
-    @Column
```
Comment (ZihengSun) [inline src/main/java/com/gw/jpa/GWUser.java:28]: Why this removal?

File: src/main/resources/application.properties [hunk 5]
```diff
@@ -12,17 +12,15 @@ server.servlet.contextPath=/Geoweaver
 logging.level.root=FATAL
 logging.level.org.springframework=FATAL
 logging.level.com.gw=FATAL
-logging.level.org.hibernate=FATAL
+logging.level.org.hibernate=DEBUG
```
Comment (ZihengSun) [inline src/main/resources/application.properties:15]: We really don't care about hibernate log in production. Can you turn this back to FATAL?

File: src/main/resources/application.properties [hunk 6]
```diff
@@ -12,17 +12,15 @@ server.servlet.contextPath=/Geoweaver
...
 
+# import the external configuration file if exists
+spring.config.import=optional:file:${HOME}/geoweaver/application.properties
```
Comment (ZihengSun) [inline src/main/resources/application.properties:18]: Is this recorded somewhere in the documentation? how do people know where to change?

**Review by ZihengSun — CHANGES_REQUESTED** [review]
LG. Some questions and comments.

File: pom.xml (hunk 1 above)
Comment (saivivek116) [reply pom.xml:85]: Those versions are latest both mysql and postgresql

File: src/main/java/com/gw/database/EnvironmentRepository.java (hunk 2 above)
Comment (saivivek116) [reply src/main/java/com/gw/database/EnvironmentRepository.java:38]: yes it works with existing h2 database

File: src/main/java/com/gw/jpa/Environment.java (hunk 3 above)
Comment (saivivek116) [reply src/main/java/com/gw/jpa/Environment.java:25]: It works, tested manually, no need for migration. Need more testing though

**Review by saivivek116 — APPROVED** [review]

**Comment by gokulprathin8** [comment]
LGTM
merging.

**Pull request merged**